src/main/java/org/snomed/snowstorm/core/data/services/AxiomConversionService.java

```diff
@@ -102,12 +102,29 @@ private AxiomRelationshipConversionService setupConversionService(String branchP
 	private Set<Relationship> mapToInternalRelationshipType(Long sourceId, Map<Integer, List<org.snomed.otf.owltoolkit.domain.Relationship>> relationships) {
 		if (relationships == null) return null;
 
-		return relationships.values().stream()
+		return relationships
+				.values()
+				.stream()
 				.flatMap(Collection::stream)
-				.map(r -> new Relationship(r.getTypeId() + "", r.getDestinationId() + "")
-						.setSourceId(sourceId != null ? sourceId.toString() : null)
-						.setGroupId(r.getGroup())
-						.setInferred(false))
+				.map(externalRelationship -> {
+					final long externalRelationshipTypeId = externalRelationship.getTypeId();
+					final long externalRelationshipDestinationId = externalRelationship.getDestinationId();
+					final Relationship internalRelationship = new Relationship(externalRelationshipTypeId + "", externalRelationshipDestinationId + "");
+					internalRelationship.setSourceId(sourceId != null ? sourceId.toString() : null);
+					internalRelationship.setGroupId(externalRelationship.getGroup());
+					internalRelationship.setInferred(false);
+
+					final org.snomed.otf.owltoolkit.domain.Relationship.ConcreteValue externalConcreteValue = externalRelationship.getValue();
+					if (externalConcreteValue != null) {
+						final String rf2Value = externalConcreteValue.getRF2Value();
+						internalRelationship.setConcreteValue(rf2Value);
+						internalRelationship.setValue(rf2Value);
```

Comment:
The code would be more robust if we could set both representations of the concrete value and clear destination and target with a single method call to the relationship object. See what you can do.

Reply:
Fixed. 

I originally debated this myself, so I'm happy with the suggestion.